File: src/funciones/lambda/lambda2.py
from typing import TypeVar

I = TypeVar(int)

def suma_n(n: int) -> I:
    suma = 0
    for i in range(1, n + 1):
        suma += i
    yield suma
d = suma_n(100)
[print(i) for i in d]

#Lambda
suman = 0
suma_nu = (lambda n: ([suman += i for i in range(1, n + 1)] return suma))
print(suma_nu(100))

File: src/funciones/lambda/lambda2.js
function suma_n(n) {
    let suma = 0;
    let con = 1;
    while (con <= n) {
        suma = suma + con;
        con++;
    }
    console.log(suma);
}
console.log(suma_n(100));


//Do-while
function sumaNP(n) {
    let suma = 0;
    let i = 1;
    do {
        suma = suma + i;
        i++;
    } while (i <= n);
    return suma;
}
console.log(sumaNP(100));

//Arrow
suma_num = (n => {
    let suma = 0;
    let con = 1;
    while (con <= n) {
        suma = suma + con;
        con++;
    }
    console.log(suma);
});
console.log(suma_num(100));